src/core/layout/qgslayoutitemelevationprofile.cpp

```diff
@@ -641,6 +641,9 @@ void QgsLayoutItemElevationProfile::paint( QPainter *painter, const QStyleOption
       layoutSize *= dotsPerMM; // output size will be in dots (pixels)
       painter->scale( 1 / dotsPerMM, 1 / dotsPerMM ); // scale painter from mm to dots
 
+      const double mapUnitsPerPixel = static_cast<double>( mPlot->xMaximum() - mPlot->xMinimum() ) / layoutSize.width();
+      rc.setMapToPixel( QgsMapToPixel( mapUnitsPerPixel ) );
```

Comment:
That will need to be added into the qgsquickelevtionprofillecanvas too I imagine?

Reply:
Done

Reply:
@nyalldawson , thanks.